pkg/config/config.go

```diff
@@ -0,0 +1,205 @@
+			c.NextProtos = []string{"http/1.1"}
+		})
+	}
+
+	o.Metrics = metricsserver.Options{
+		BindAddress:   cfg.Metrics.BindAddress,
+		SecureServing: cfg.Metrics.SecureServing != nil && *cfg.Metrics.SecureServing,
+		TLSOpts:       tlsOpts,
+	}
+
+	// Set webhook server options
+	if cfg.Webhook.Port != nil {
+		o.WebhookServer = webhook.NewServer(webhook.Options{
+			Port:    *cfg.Webhook.Port,
+			Host:    cfg.Webhook.Host,
+			TLSOpts: tlsOpts,
+		})
+	}
+
+	// Set health probe bind address
+	o.HealthProbeBindAddress = cfg.Health.HealthProbeBindAddress
+
+	// Set leader election
+	if cfg.LeaderElection != nil {
+		if cfg.LeaderElection.LeaderElect != nil {
+			o.LeaderElection = *cfg.LeaderElection.LeaderElect
+		}
+		o.LeaderElectionResourceLock = cfg.LeaderElection.ResourceLock
+		o.LeaderElectionNamespace = cfg.LeaderElection.ResourceNamespace
+		o.LeaderElectionID = cfg.LeaderElection.ResourceName
+		o.LeaseDuration = &cfg.LeaderElection.LeaseDuration.Duration
+		o.RenewDeadline = &cfg.LeaderElection.RenewDeadline.Duration
+		o.RetryPeriod = &cfg.LeaderElection.RetryPeriod.Duration
+	}
+
+	// Set controller concurrency if specified
+	if cfg.Controller != nil && len(cfg.Controller.GroupKindConcurrency) > 0 {
+		if o.Controller.GroupKindConcurrency == nil {
+			o.Controller.GroupKindConcurrency = make(map[string]int)
+		}
+		for gk, concurrency := range cfg.Controller.GroupKindConcurrency {
+			o.Controller.GroupKindConcurrency[gk] = concurrency
+		}
+	}
+}
+
+// Load loads configuration from file and returns controller Options and Configuration.
+// If configFile is empty, default configuration is used.
+func Load(scheme *runtime.Scheme, configFile string, enableHTTP2 bool) (ctrl.Options, configapi.Configuration, error) {
+	options := ctrl.Options{
+		Scheme: scheme,
+	}
+
+	cfg := configapi.Configuration{}
+
+	if configFile == "" {
+		// Apply defaults
+		scheme.Default(&cfg)
+	} else {
+		// Load from file
+		if err := fromFile(configFile, scheme, &cfg); err != nil {
+			return options, cfg, err
+		}
+	}
+
+	// Validate configuration
+	if errs := validate(&cfg); len(errs) > 0 {
+		return options, cfg, fmt.Errorf("invalid configuration: %v", errs.ToAggregate())
+	}
+
+	// Apply configuration to options
+	addTo(&options, &cfg, enableHTTP2)
+
+	return options, cfg, nil
+}
+
+// IsCertManagementEnabled returns true if certificate management is enabled.
+// Returns true by default if not explicitly disabled.
+func IsCertManagementEnabled(cfg *configapi.Configuration) bool {
+	if cfg.CertManagement == nil || cfg.CertManagement.Enable == nil {
+		return true // Enabled by default
+	}
+	return *cfg.CertManagement.Enable
+}
+
+// GetWebhookServiceName returns the webhook service name from configuration.
+func GetWebhookServiceName(cfg *configapi.Configuration) string {
+	if cfg.CertManagement != nil && cfg.CertManagement.WebhookServiceName != "" {
+		return cfg.CertManagement.WebhookServiceName
+	}
+	return "kubeflow-trainer-controller-manager"
```

Comment:
Do you need this since we enforce defaults here: https://github.com/kubeflow/trainer/blob/aaa3c07d4d3c43d50e2ffcaca709c4530bc2b982/pkg/apis/config/v1alpha1/defaults.go